src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -255,13 +244,37 @@ private async Task<int> PromoteBuildAsync(Build build, List<Channel> targetChann
                 return Constants.ErrorCode;
             }
 
+            // Construct the templateParameters and queue time variables. 
+            // Publishing v2 uses variables and v3 uses parameters, so just use the same values for both.
+            var promotionPipelineVariables = new Dictionary<string, string>
+            {
+                { "BarBuildId", build.Id.ToString() },
+                { "PublishingInfraVersion", _options.PublishingInfraVersion.ToString() },
+                { "PromoteToChannelIds", string.Join(",", targetChannels.Select(tch => tch.Id)) },
+                { "EnableSigningValidation", _options.DoSigningValidation.ToString() },
+                { "SigningValidationAdditionalParameters", _options.SigningValidationAdditionalParameters },
+                { "EnableNugetValidation", _options.DoNuGetValidation.ToString() },
+                { "EnableSourceLinkValidation", _options.DoSourcelinkValidation.ToString() },
+                { "PublishInstallersAndChecksums", _options.PublishInstallersAndChecksums.ToString() },
+                { "SymbolPublishingAdditionalParameters", _options.SymbolPublishingAdditionalParameters },
+                { "ArtifactsPublishingAdditionalParameters", _options.ArtifactPublishingAdditionalParameters }
+            };
+
+            if (_options.DoSDLValidation)
+            {
+                promotionPipelineVariables.Add("EnableSDLValidation", _options.DoSDLValidation.ToString());
+                promotionPipelineVariables.Add("SDLValidationCustomParams", _options.SDLValidationParams);
+                promotionPipelineVariables.Add("SDLValidationContinueOnError", _options.SDLValidationContinueOnError);
+            }
+
             int azdoBuildId = await azdoClient.StartNewBuildAsync(build.AzureDevOpsAccount,
                 promotionPipelineInformation.project,
                 promotionPipelineInformation.pipelineId,
                 arcadeSDKSourceBranch,
                 arcadeSDKSourceSHA,
-                queueTimeVariables)
-                .ConfigureAwait(false);
+                promotionPipelineVariables,
```

Comment:
might merit a comment why you're passing the same variable twice

Reply:
That was what https://github.com/dotnet/arcade-services/pull/1351/files#diff-aa5a491937907bb6146e94e0ab683fd3R247-R248 was supposed to convey, but it's the second comment about it so I can clearly be more clear about it.